subcommand/mesh-init/command.go

```diff
@@ -1,89 +1,61 @@
 package meshinit
 
 import (
-	"encoding/json"
-	"flag"
 	"fmt"
 	"io/ioutil"
 	"os"
 	"os/exec"
 	"path"
-	"strconv"
-	"strings"
 	"sync"
 	"time"
 
 	"github.com/cenkalti/backoff/v4"
 	"github.com/hashicorp/consul-ecs/awsutil"
+	"github.com/hashicorp/consul-ecs/config"
 	"github.com/hashicorp/consul/api"
 	"github.com/hashicorp/go-hclog"
 	"github.com/mitchellh/cli"
 )
 
 const (
-	flagEnvoyBootstrapDir    = "envoy-bootstrap-dir"
-	flagPort                 = "port"
-	flagTags                 = "tags"
-	flagMeta                 = "meta"
-	flagServiceName          = "service-name"
-	flagUpstreams            = "upstreams"
-	flagChecks               = "checks"
-	flagHealthSyncContainers = "health-sync-containers"
-
 	envoyBoostrapConfigFilename = "envoy-bootstrap.json"
 )
 
 type Command struct {
-	UI cli.Ui
-
-	flagEnvoyBootstrapDir    string
-	flagPort                 int
-	flagServiceName          string
-	flagTags                 string
-	flagMeta                 string
-	flagUpstreams            string
-	flagChecks               string
-	flagHealthSyncContainers string
-
-	flagSet *flag.FlagSet
-	once    sync.Once
+	UI     cli.Ui
+	config *config.Config
+	once   sync.Once
+	log    hclog.Logger
 }
 
 func (c *Command) init() {
-	c.flagSet = flag.NewFlagSet("", flag.ContinueOnError)
-	c.flagSet.StringVar(&c.flagEnvoyBootstrapDir, flagEnvoyBootstrapDir, "",
-		"Directory for Envoy startup files. The envoy-bootstrap.json and the consul-ecs binary are written here.")
-	c.flagSet.IntVar(&c.flagPort, flagPort, 0, "Port service runs on")
-	c.flagSet.StringVar(&c.flagUpstreams, flagUpstreams, "", "Upstreams in form <name>:<port>,...")
-	c.flagSet.StringVar(&c.flagChecks, flagChecks, "", "List of Consul checks in JSON form")
-	c.flagSet.StringVar(&c.flagHealthSyncContainers, flagHealthSyncContainers, "",
-		"A comma separated list of container names that need Consul TTL checks")
-	c.flagSet.StringVar(&c.flagTags, flagTags, "", "Tags for the Consul service as a comma separated string")
-	c.flagSet.StringVar(&c.flagMeta, flagMeta, "", "Metadata for the Consul service as a JSON string")
-	c.flagSet.StringVar(&c.flagServiceName, flagServiceName, "",
-		"Name of the service that will be registered with Consul. If not provided, the task family will be used as the service name.")
+	c.log = hclog.New(nil)
 }
 
-func (c *Command) Run(args []string) int {
+func (c *Command) Run(_ []string) int {
 	c.once.Do(c.init)
-	if err := c.flagSet.Parse(args); err != nil {
+
+	config, err := config.FromEnv()
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("invalid config: %s", err))
 		return 1
 	}
-	if c.flagEnvoyBootstrapDir == "" {
-		c.UI.Error(fmt.Sprintf("-%s must be set", flagEnvoyBootstrapDir))
+	c.config = config
+
+	if c.config.Mesh.BootstrapDir == "" {
```

Comment:
What do you think about moving this check to JSON Schema? I like the idea of doing as many validations there as possible.